lib_eio_linux/getaddrinfo_stubs.c

```diff
@@ -0,0 +1,187 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*             Xavier Leroy, projet Cristal, INRIA Rocquencourt,          */
+/*             Christiano Haesbaert, Tarides                              */
+/*   Copyright 2004 Institut National de Recherche en Informatique et     */
+/*     en Automatique.                                                    */
+/*   Copyright 2022 Tarides                                               */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+#include <sys/types.h>
+#include <sys/socket.h>
+
+#include <errno.h>
+#include <netdb.h>
+
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/alloc.h>
+#include <caml/unixsupport.h>
+#include <caml/socketaddr.h>
+
+static value caml_unix_cst_to_constr(int n, int *tbl, int size, int deflt)
```

Comment:
You could avoid this duplication by borrowing the `extern` declaration in `cst2constr.h`

Reply:
fixed, somehow my brain remembered the function being static in first place.